GiEnJul.Test/Controllers/GiverControllerTest.cs

```diff
@@ -0,0 +1,95 @@
+﻿using GiEnJul.Features;
+using GiEnJul.Test;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace GiEnJul.Controllers.Tests
+{
+    public class GiverControllerTest
+    {
+        private Mock<IGiverRepository> mockGiverRepo { get; set; }
+        private Mock<ILogger> mockLog;
+        private GiverController _controller;
+
+
+        public GiverControllerTest()
+        {
+            mockGiverRepo = new Mock<IGiverRepository>();
+            mockLog = new Mock<ILogger>();
+            _controller = new GiverController(mockGiverRepo.Object, mockLog.Object);
+        }
+
+
+        /// <summary>
+        /// Validate model using DataAnnotations such as [Required] etc.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns>List of ValidationErrors or empty list</returns>
+        private IList<ValidationResult> ValidateModel(object model)
+        {
+            var validationResults = new List<ValidationResult>();
+            var ctx = new ValidationContext(model, null, null);
+            Validator.TryValidateObject(model, ctx, validationResults, true);
+            return validationResults;
+        }
+
+        public static IEnumerable<object[]> GetInvalidGivers()
+        {
+            yield return new object[] { new Models.Giver("Location") }; //Missing members
+            yield return new object[] { new Models.Giver("Stavanger") { FullName = "Test Testesen", Email = "Test@Testesen.com",    PhoneNumber = "12345678", MaxRecievers = 0 } }; //Invalid MaxRecivers
+            yield return new object[] { new Models.Giver("Stavanger") { FullName = "",              Email = "Test@Testesen.com",    PhoneNumber = "12345678", MaxRecievers = 5 } }; //Invalid Name
+            yield return new object[] { new Models.Giver("Stavanger") { FullName = "Test Testesen", Email = "Invalid Mail",         PhoneNumber = "12345678", MaxRecievers = 5 } }; //Invalid Mail
+            yield return new object[] { new Models.Giver("Stavanger") { FullName = "Test Testesen", Email = "Test@Testesen.com",    PhoneNumber = "Invalid ", MaxRecievers = 5 } }; //Invalid PhoneNumber
+        }
+
+        [Theory]
+        [MemberData(nameof(GetInvalidGivers))]
+        public void ValidateInvalidGiverModels(Models.Giver model)
+        {
+            Assert.NotEmpty(ValidateModel(model));
+        }
+
+        [Fact]
+        public void ValidateValidGiverModel()
+        {
+            Assert.Empty(ValidateModel(new Models.Giver("Stavanger") { FullName = "Test Testesen", Email = "Test@Testesen.com", PhoneNumber = "12345678", MaxRecievers = 1 }));
+        }
```

Comment:
@RonHaland Should these test be seperated from GiverController to another Test-class?
They test the Dataannotations for Models.Giver.

Reply:
Yeah, probably